modules/twinkleblock.js

```diff
@@ -984,13 +983,52 @@ Twinkle.block.blockPresetsInfo = {
 		reason: '{{zombie proxy}}',
 		sig: null
 	},
+
+	// Begin partial block templates, accessed in Twinkle.block.blockGroupsPartial
 	'uw-pblock': {
 		autoblock: true,
 		expiry: '24 hours',
 		nocreate: false,
 		pageParam: false,
 		reasonParam: true,
 		summary: 'You have been [[WP:PB|partially blocked]] from certain areas of the encyclopedia'
+	},
+	'uw-pblockindef': {
+		autoblock: true,
+		expiry: 'infinity',
+		forRegisteredOnly: true,
+		nocreate: false,
+		pageParam: false,
+		reasonParam: true,
+		summary: 'You have been indefinitely [[WP:PB|partially blocked]] from certain areas of the encyclopedia'
+	},
+	'uw-ewpblock': {
+		autoblock: true,
+		expiry: '24 hours',
+		nocreate: false,
+		pageParam: false,
+		reasonParam: true,
+		reason: '[[WP:Edit warring|Edit warring]]',
+		summary: 'You have been [[WP:PB|blocked]] from editing certain areas of the encyclopedia to prevent further [[WP:DE|disruption]] caused by your engagement in an [[WP:EW|edit war]]'
+	},
+	'uw-acpblock': {
+		autoblock: true,
+		expiry: '24 hours',
```

Comment:
set to infinity.

Reply:
Couldn't this be applied to IPs as well @QEDK?  Probably the thing to do is create two presets, `forRegisteredOnly` and `forAnonOnly` with infinity and 48(?) hours as the defaults.

Reply:
Yep, that's a better idea.

Reply:
What's another template among friends, eh?  I suppose there's no need for the temporary one to be only anon-only.